server/service/hosts.go

```diff
@@ -1029,17 +1030,30 @@ func (svc *Service) getHostDetails(ctx context.Context, host *fleet.Host, opts f
 			if !ac.MDM.WindowsEnabledAndConfigured {
 				break
 			}
+
 			if license.IsPremium(ctx) {
-				hde, err := svc.ds.GetMDMWindowsBitLockerStatus(ctx, host)
+				// we include disk encryption status only for premium so initialize it to default struct
+				host.MDM.OSSettings.DiskEncryption = fleet.HostMDMDiskEncryption{}
+				// ensure host mdm info is loaded (we don't know if our caller populated it)
+				err := svc.ensureHostMDMInfo(ctx, host)
```

Comment:
this is a good improvement, is there anything we could do for the check to be done? otherwise we're in a similar spot, right?

eg (not necessarily suggesting this) could we make `host.MDMinfo` only accessible by a function that does the check?

Reply:
Yes, we're still relying on the caller for now to know that this method exists and that it should be used. For the broader fix, I think we'd probably need to start returning an error from the helper methods that are currently just returning booleans instead of defaulting to false when nil. That will make the calls more verbose, especially given how the booleans are being used currently, but we can look for a pattern that makes things less painful for the caller when it needs to assess several different booleans.

Reply:
gotcha, yeah if this is not going to be the pattern for the broader fix, then LGTM!